src/near-rpc-provider.ts

```diff
@@ -154,10 +162,42 @@ export class NearRpcProvider extends JsonRpcProvider {
     switch (method) {
       case 'getBlockNumber':
         const blockResponse = await this.send<BlockRpcResponse>('block', { finality: 'final' })
-
         return blockResponse.header.height
+      case 'getBalance':
+        return this._internalGetBalance(method, params)
       default:
         return super.perform(method, params)
     }
   }
+
+  _getAddress(addressOrName: string | Promise<string>): Promise<string> {
+    return Promise.resolve(addressOrName)
+  }
+
+  private async _internalGetBalance(method: string, params: Record<string, any>): Promise<BigNumber> {
+    const defaultParams = {
+      request_type: 'view_account' as const,
+      finality: 'final' as const,
+    }
+
+    const getBalanceParams: GetBalanceParams = {
+      ...defaultParams,
+      ...params,
+      account_id: params.address,
+    }
+
+    validateGetBalanceParams(getBalanceParams)
```

Comment:
we don't need to validate the params. We just want to get the balance from an account. this function `_internalGetBalance` just need to received the address and that's it. Then build the query to get the balance and return a bignumber